linker/shared/src/main/scala/org/scalajs/linker/backend/javascript/Printers.scala

```diff
@@ -747,9 +747,13 @@ object Printers {
     protected def print(ident: Ident): Unit =
       printEscapeJS(ident.name)
 
+    protected def print(ident: DelayedIdent): Unit =
+      printEscapeJS(ident.resolveName())
```

Comment:
We seem to be running into a similar issue than I had when working on the post transforms (in the version where we still had a transient tree).

We use Javascript tree printing for 2 things:
- Actual printing
- Showing a tree for debugging

In the latter case, this line is a problem: It might in case we attempt to show an unexpected tree in an exception / assertion message (because we have not generated names yet).

I'm not sure if it is worth adding options to Printers to deal with this :-/

Reply:
We could probably define a subclass `JSTreeShowPrinter`, extending `JSTreePrinter` and overriding this method to show `<delayedIdent>` instead. Would that make sense?

The subclass should be better than an option on the JVM, because on the happy it's never going to be used and therefore inline caching will essentially consider it dead code.

Reply:
I think the proposed approach in the last commit looks good (modulo some details I've commented on).

I'm unsure about what the best commit structure is for it though. I feel there's a lot in the minify commit already, so maybe before, or even a separate PR (with the improved interface and a smoke test)?